tensorboard/plugins/interactive_inference/witwidget/notebook/jupyter/js/lib/wit.js

```diff
@@ -118,14 +123,19 @@ var WITView = widgets.DOMWidgetView.extend({
       this.model.set('get_eligible_features', i);
       this.touch();
     });
-
     this.inferMutantsCounter = 0;
     this.view_.addEventListener('infer-mutants', (e) => {
       e.detail['infer_mutants_counter'] = this.inferMutantsCounter++;
       this.model.set('infer_mutants', e.detail);
       this.mutantFeature = e.detail.feature_name;
       this.touch();
     });
+    this.computeDistanceCounter = 0;
+    this.view_.addEventListener('compute-custom-distance', (e) => {
+      e.detail['compute_distance_counter'] = this.computeDistanceCounter++;
```

Comment:
this feels quite odd and not clean. Would it be possible to use data binding instead?

Reply:
The counter is there as it is in the case of infer-mutants above. We use data binding for what is passed ('compute_custom_distance' is set to e.detail). However, we only pass the example id back to the python side instead of the full example since examples are already in sync between the backend and frontend. In this case, if someone requests the same example again after updating some features, the change on bound dict may not trigger since it is the same dict. We force trigger it by incrementing the counter.

Reply:
Sorry but I must confess I am very lost. Even if this logic is correct, I do think this has readability problem.

AFAICT, the event is fired from JavaScript side. When firing the event, the object does not contain the property `compute_distance_counter`. 1. how is it being used by owner of the event object? 2. where does the Python come into play?

Reply:
The dashboard launches this event from js side by passing some parameters (example index, callback ind etc.).  "compute_custom_distance" is sync variable between js and python sides. This synchronization is managed by jupyter widget framework. When we call this.model.set('compute_custom_distance', e.detail), ideally jupyter should recognize the change in "compute_custom_distance" and trigger distance computation. James and I found out previously (for infer-mutants) that if someone edits an example then fires this event again from the dashboard, since e.detail does not contain the contents of the example but only it's index, jupyter thinks nothing has changed and model.set does not trigger distance computation. James solved it by adding a counter so that we make sure the serialized dictionary always has one field that changes between calls and jupyter triggers the sync.
Since this is a jupyter specific issue, we did not include this in the event but append it in wit.js related to jupyter.